### PR TITLE
feat: add SPDX headers

### PR DESCRIPTION
**Pull request by luka-nextcloud**
Resolves: #1325

### CODE REVIEW

**Comment by juliushaertl** [comment]
We also need:
- [ ] Add a github action to check this https://github.com/nextcloud/.github/blob/master/workflow-templates/reuse.yml
- [ ] Make sure that the check passes
- [ ] Add an AUTHORS file (see https://github.com/nextcloud/text/pull/5738)
- [ ] Have a reuse config file for cases where we cannot define the license in the headers (see https://github.com/nextcloud/whiteboard/pull/69/files)

File: appinfo/routes.php [hunk 1]
```diff
@@ -1,5 +1,10 @@
 <?php
 
+/**
+ * SPDX-FileCopyrightText: 2013 Nextcloud GmbH and Nextcloud contributors
```
Comment (juliushaertl) [inline appinfo/routes.php:4]: Need to double check if that is OK, or if we need to have a different copyright header since 2013 was before Nextcloud

Comment (luka-nextcloud) [reply appinfo/routes.php:4]: There is copyright from first commit: https://github.com/nextcloud/notes/commit/1c4ae20dad288b5691e87a1fabb5e944071ca782#diff-6301937af1cc1575e9e18ecd27c4a51f0d3cf9ce12e59a14709f626dfa3ef952R2-R6

Comment (juliushaertl) [reply appinfo/routes.php:4]: Maybe @AndyScherzinger can easily tell if we can attribute that as `Nextcloud GmbH and Nextcloud contributors`or if we need to keep Bernhard explicitly in there

Comment (AndyScherzinger) [reply appinfo/routes.php:4]: I'd say keep him explicitly there. Given code existing before Nextcloud has been founded, Nextcloud cannot be a Copyright holder in any way.
So for anything oder than 2016, best to keep the original copyright holder Infos in additionally.

File: appinfo/info.xml [hunk 2]
```diff
@@ -1,4 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
+<!--
+  - SPDX-FileCopyrightText: 2013-2024 Nextcloud GmbH and Nextcloud contributors
```
Comment (AndyScherzinger) [inline appinfo/info.xml:3]: Should be changed to 
SPDX-FileCopyrightText: 2016-2024 Nextcloud GmbH and Nextcloud contributors
SPDX-FileCopyrightText: 2013 Bernhard Posselt <nukeawhale@gmail.com>

File: README.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+<!--
+  - SPDX-FileCopyrightText: 2019-2024 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
 # Notes
```
Comment (AndyScherzinger) [inline README.md:6]: Also please add the status badge here via:
```
[![REUSE status](https://api.reuse.software/badge/github.com/nextcloud/notes)](https://api.reuse.software/info/github.com/nextcloud/notes)
```

**Comment by AndyScherzinger** [comment]
Changes look good to me reuse/spdx-wise. Just 2 minor comments else 👍

**Review by AndyScherzinger — APPROVED** [review]

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**